src/components/RoomView/customHooks/OnValueCardsRef.ts

```diff
@@ -15,109 +14,50 @@ function OnValueCardsRef() {
-                    teamId: card.teamId,
-                  }),
-              );
-              const values = Object.values(wordsWithTeamIds);
-              dispatch(setWordsInGame(values));
-            }
-          }
-        });
-        get(teamTwoSpymasterRef).then(async (snapshot) => {
-          if (snapshot.exists()) {
-            let spymaster = snapshot.val();
-            if (spymaster.playerId === playerId) {
-              console.log('setting spy board...');
-              //get set of cards with team ids from backend and set spymaster words
-              let wordsWithTeamIds = {} as WordsWithTeamIdsObj;
-              let spyWords = await axios.get(`/api/card/get25/forRoom/${roomId}`);
-              spyWords.data.forEach(
-                (card: CardObj) =>
-                  (wordsWithTeamIds[card.id] = {
-                    id: card.id,
-                    isVisibleToAll: card.isVisibleToAll,
-                    wordString: card.word.word,
-                    word: card.word,
-                    wordId: card.wordId,
-                    boardId: card.boardId,
-                    teamId: card.teamId,
-                  }),
-              );
-              const values = Object.values(wordsWithTeamIds);
-              dispatch(setWordsInGame(values));
-            }
-          }
-        });
-        get(teamOneOperativesRef).then((snapshot) => {
-          if (snapshot.exists()) {
-            let operatives = snapshot.val();
-            let operativesIds = Object.keys(operatives);
-            if (operativesIds.includes(playerId)) {
-              console.log('setting opertive board...');
-              //update our redux to reflect that
-              const cardsFromSnapshot = cardSnapshot.val();
-              const values = Object.values(cardsFromSnapshot);
-              dispatch(setWordsInGame(values));
-            }
-          }
-        });
-        get(teamTwoOperativesRef).then((snapshot) => {
-          if (snapshot.exists()) {
-            let operatives = snapshot.val();
-            let operativesIds = Object.keys(operatives);
-            if (operativesIds.includes(playerId)) {
-              console.log('setting operative board...');
-              //update our redux to reflect that
-              const cardsFromSnapshot = cardSnapshot.val();
-              const values = Object.values(cardsFromSnapshot);
-              dispatch(setWordsInGame(values));
-            }
-          }
-        });
-        get(nestedPlayerRef).then((snapshot) => {
-          if (snapshot.exists()) {
-              console.log('setting spectator board...');
-              const cardsFromSnapshot = cardSnapshot.val();
-              const values = Object.values(cardsFromSnapshot);
-              dispatch(setWordsInGame(values));
-          }
-        });
+        if (playerIsSpymaster) {
+          console.log('setting spy board...');
+          off(cardsRef); // don't listen to this listener anymore.
```

Comment:
just realized - I think we can have this only be a single line rather than under this `get` & then the `else`. It should appear directly under the line 28 `if (cardSnapshot.exists()) ` because as soon as we get that one large cards snapshot, we now no longer need to listen